src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -132,9 +132,9 @@ public ReadOnlySequence(ReadOnlyMemory<T> readOnlyMemory)
             }
             else if (typeof(T) == typeof(char))
             {
-                // TODO: MemoryMarshal.TryGetString -- doesn't exist?
-                // https://github.com/dotnet/corefx/issues/27451
-                MemoryExtensions.TryGetString(((ReadOnlyMemory<char>)(object)readOnlyMemory), out string text, out int start, out length);
+                // At this point, ReadOnlyMemory is backed by a string object, so TryGetString shouldn't fail
+                bool result = MemoryMarshal.TryGetString(((ReadOnlyMemory<char>)(object)readOnlyMemory), out string text, out int start, out length);
+                Debug.Assert(result);
```

Comment:
The error handling should be consistent. If it using assert here and throw a few lines below. Either both should be asserts or both should be throwing exceptions.

Reply:
change it to `else if` and let the `else` exception handle it?

Reply:
Something like this?

```C#
...
else if (typeof(T) == typeof(char) && MemoryMarshal.TryGetString(((ReadOnlyMemory<char>)(object)readOnlyMemory), out string text, out int start, out length))
{
    _sequenceStart = new SequencePosition(text, ReadOnlySequence.StringToSequenceStart(start));
    _sequenceEnd = new SequencePosition(text, ReadOnlySequence.StringToSequenceEnd(start + length));
}
else
{
    ThrowHelper.ThrowInvalidOperationException();
...
}
```

Reply:
Yes, if it compiles...

Reply:
Is the `typeof(T) == tyepof(char)` test necessary?

Reply:
> Yes, if it compiles...

```C#
else if (typeof(T) == typeof(char))
{
    if (MemoryMarshal.TryGetString(((ReadOnlyMemory<char>)(object)readOnlyMemory), out string text, out int start, out length))
    {
        _sequenceStart = new SequencePosition(text, ReadOnlySequence.StringToSequenceStart(start));
        _sequenceEnd = new SequencePosition(text, ReadOnlySequence.StringToSequenceEnd(start + length));
    }
}
else
```

Reply:
`(ReadOnlyMemory<char>)(object)readOnlyMemory` would throw invalid cast exception without it.

Reply:
Removes the code from the generic output when its not `char`

Reply:
> Yes, if it compiles...

This is missing the error handling again. With error handling, it should be:

```
else if (typeof(T) == typeof(char))
{
    if (!MemoryMarshal.TryGetString(((ReadOnlyMemory<char>)(object)readOnlyMemory), out string text, out int start, out length))
         ThrowHelper.ThrowInvalidOperationException();

    _sequenceStart = new SequencePosition(text, ReadOnlySequence.StringToSequenceStart(start));
    _sequenceEnd = new SequencePosition(text, ReadOnlySequence.StringToSequenceEnd(start + length));
}
```